### PR TITLE
Add Redfish to Power Management

### PR DESCRIPTION
**Pull request by Loquacity**
# Description

Add Redfish protocol to power management chapter in client cfg, and update Ref Guide in SD-Provisioning. Also some drive-by editing on the `Enabling Power Management` procedure.

# Target branches

Which product version does this PR apply to?

- [ ] Manager-4.0
- [ ] Manager-4.1
- [x] Manager-4.2
- [x] Uyuni

# Links

Fixes https://github.com/SUSE/spacewalk/issues/12767

### CODE REVIEW

**Comment by Loquacity** [comment]
@mcalmer in the webui, it basically says that Redfish isn't supported. Does this need to be changed?

image

**Comment by mcalmer** [comment]
> @mcalmer in the webui, it basically says that Redfish isn't supported. Does this need to be changed?
> 
> image

Good catch. I have to change that message.

**Comment by mcalmer** [comment]
> > @mcalmer in the webui, it basically says that Redfish isn't supported. Does this need to be changed?
> > image
> 
> Good catch. I have to change that message.

What about the following text:

    By default only the tested and supported power management types are listed, but others may work. To enable other power 
    management types override the "java.power_management.types" option in rhn.conf.

**Comment by Loquacity** [comment]
> 
> ```
> By default only the tested and supported power management types are listed, but others may work. To enable other power 
> management types override the "java.power_management.types" option in rhn.conf.
> ```

```
The power management types listed are tested and supported, but other types might work. To enable other power management types edit the "java.power_management.types" option in rhn.conf.
 ```

**Comment by mcalmer** [comment]
> > ```
> > By default only the tested and supported power management types are listed, but others may work. To enable other power 
> > management types override the "java.power_management.types" option in rhn.conf.
> > ```
> 
> ```
> The power management types listed are tested and supported, but other types might work. To enable other power 
> management types edit the "java.power_management.types" option in rhn.conf.
> ```

As soon as you modify the var, the types are also listed. But I hope we can ignore this detail and nobody will demand for support when they add a new type and it does not work.

File: modules/client-configuration/pages/power-management.adoc [hunk 1]
```diff
@@ -3,12 +3,19 @@
 
 You can power on, power off, and reboot clients using the {productname} {webui}.
 
-This feature uses the IPMI protocol and is managed using a Cobbler profile.
-The client must be IPMI-enabled, and configured to use a power management server.
-To use any power management functionality, IPMI configuration details must be added to {productname}.
+This feature uses either the IPMI or Redfish protocol and is managed using a Cobbler profile.
+The selected client must have the protocol enabled, and be configured to use a power management server.
```
Comment (mcalmer) [inline modules/client-configuration/pages/power-management.adoc:7]:
```suggestion
The selected client must have a power management controller supporting one of these protocols.
```

File: modules/client-configuration/pages/power-management.adoc [hunk 2]
```diff
@@ -3,12 +3,19 @@
...
 
-Navigate to menu:Systems[Systems List], select the client you want to manage, and navigate to the menu:Provisioning[Power Management] tab.
-Complete the details for the power management server, and click the appropriate button for the action to take, or click btn:[Save only] to save the details without taking any action.
+For Redfish, ensure you have a valid SSL connection between the client and the {productname} Server.
```
Comment (mcalmer) [inline modules/client-configuration/pages/power-management.adoc:9]:
```suggestion
For Redfish, ensure you can establish a valid SSL connection between the client and the {productname} Server.
```

File: modules/reference/pages/systems/system-details/sd-provisioning.adoc [hunk 3]
```diff
@@ -21,8 +21,9 @@ For more information about autoinstallation, see xref:client-configuration:autoi
 
 This section allows you to power on, power off, and reboot the selected client.
 
-This feature uses the IPMI protocol and is managed using a Cobbler profile.
-The selected client must be IPMI-enabled, and configured to use a power management server.
+This feature uses either the IPMI or Redfish protocol and is managed using a Cobbler profile.
+The selected client must have the protocol enabled, and be configured to use a power management server.
```
Comment (mcalmer) [inline modules/reference/pages/systems/system-details/sd-provisioning.adoc:25]:
```suggestion
The selected client must have a power management controller supporting one of these protocols
```

File: modules/reference/pages/systems/system-details/sd-provisioning.adoc [hunk 4]
```diff
@@ -21,8 +21,9 @@ For more information about autoinstallation, see xref:client-configuration:autoi
...
+For Redfish, ensure you have a valid SSL connection between the client and the {productname} Server.
```
Comment (mcalmer) [inline modules/reference/pages/systems/system-details/sd-provisioning.adoc:26]:
```suggestion
For Redfish, ensure you can establish a valid SSL connection between the client and the {productname} Server.
```

**Review by mcalmer — CHANGES_REQUESTED** [review]
Added some suggestions.

Mainly Power Management is not only a protocol which you can install on the client.
It is a separate mini computer which control the main server. It has an own IP address, OS and hardware. It is called "Controller" and I think this makes it better understandable to talk about the controller instead of a protocol.

File: modules/reference/pages/systems/system-details/sd-provisioning.adoc [hunk 5]
```diff
@@ -21,8 +21,9 @@ For more information about autoinstallation, see xref:client-configuration:autoi
 
 This section allows you to power on, power off, and reboot the selected client.
 
-This feature uses the IPMI protocol and is managed using a Cobbler profile.
-The selected client must be IPMI-enabled, and configured to use a power management server.
+This feature uses either the IPMI or Redfish protocol and is managed using a Cobbler profile.
+The selected client must have a power management controller supporting one of these protocols
```
Comment (keichwa) [inline modules/reference/pages/systems/system-details/sd-provisioning.adoc:25]:
```suggestion
The selected client must have a power management controller supporting one of these protocols.
```

**Review by keichwa — APPROVED** [review]

**Review by mcalmer — APPROVED** [review]

**Pull request merged**